flower-core/src/commonMain/kotlin/com/hadiyarajesh/flower_core/ApiResponse.kt

```diff
@@ -53,4 +57,14 @@ data class ApiSuccessResponse<T>(
  */
 class ApiEmptyResponse<T> : ApiResponse<T>()
 
-data class ApiErrorResponse<T>(val errorMessage: String, val statusCode: Int) : ApiResponse<T>()
+//data class ApiErrorResponse<T>(val errorMessage: String, val statusCode: Int) : ApiResponse<T>()
+data class ApiErrorResponse<T>(
+    val errorMessage: ErrorMessage,
+    val statusCode: HttpStatusCode
+) : ApiResponse<T>()
+
+@JvmInline
+value class ErrorMessage(val message: String)
+
+@JvmInline
+value class HttpStatusCode(val code: Int)
```

Comment:
Calling 

```kotlin
status.errorMessage.message
// and
status.statusCode.code
```

sounds a little bit like double trouble.

How about something like this https://github.com/ktorio/ktor/blob/main/ktor-http/common/src/io/ktor/http/HttpStatusCode.kt

Reply:
Creating a data class for this seemed overkill to me. that's why to take advabtages of primitives, and stay in sync with our current apporach (error message and code), I've just created those value classes, so it's easy to understand what value it takes. 

Below code add one extra porperty call, but I think it conveys that while developing library, we should pass Htpp status code and not random string to this field. Same goes for error message.

```kotlin
status.errorMessage.message
// and
status.statusCode.code
```

Reply:
I'm not a fan of the extra property call.
I think it's kind of confusing for people, they probably won't understand why it's has an extra property with just one value.

I suggest to either go with fully primitive types (so just String and Int) or with a class holding both.

The user doesn't create an ErrorMessage anyway, this is internal only.

Reply:
That's also correct
Ok so I'll change it to primitives
But you'll need to make it compatible with the latest ktorfit as when I tried to update it, it created an error in `ResponsConverter` which we're not using though.